### PR TITLE
Usr sse2

### PR DESCRIPTION
**Pull request by rread**
I want to be able to call RDM from a script, so I needed it to exit when used as a command line tool.
Here are a some of cleanups for the command line. Now if any argument is passed it runs in command line mode only and exits. 

When no arguments are passed it runs as a normal app.

### CODE REVIEW

File: cmdline.h [hunk 1]
```diff
@@ -1,3 +1,3 @@
 
 
-int cmdline_main(int argc, const char * argv[]);
+int cmdline_main(int argc, char * const *argv);
```
Comment (usr-sse2) [inline cmdline.h:3]: What's the purpose of this change?

File: main.mm [hunk 2]
```diff
@@ -7,33 +7,19 @@
 #import "cmdline.h"
 
 
-int main(int argc, const char* argv[])
+int main(int argc, char* argv[])
```
Comment (usr-sse2) [inline main.mm:10]: Why did `const` disappear?

**Comment by usr-sse2** [comment]
Using getopt seems good. Actually exiting worked in 2.2, but then I've accidentally broken it.

File: main.mm (hunk 2 above)
Comment (rread) [reply main.mm:10]: The arguments passed to main() are not const, so there was no need for this const in the first place. I dropped it to make argv compatible with the prototype change in cmdline_main().

File: cmdline.h (hunk 1 above)
Comment (rread) [reply cmdline.h:3]: This matches the type required by getopt_long(), so putting it here avoids having to use a typecast.

**Comment by rread** [comment]
And getopt replaces a lot of code full of unchecked pointer references, which was really the main reason. Hopefully changing a couple of command line options is a reasonable tradeoff for a net reduction in code.

**Comment by rread** [comment]
Oops, I forgot github will add patches to a PR when pushed to the same branch. I was doing a bit of refactoring on found another bug - list displays was broken when more than one display is connected. I can revert this off the branch if you want it in a new PR.
